src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+						shouldContinue = false;
+					else
+						shouldContinue = true;
+
+					if (cursorRect.Y - innerScrollValue < topLayoutGuide)
+						move = cursorRect.Y - innerScrollValue - (nfloat)topLayoutGuide;
+					else if (cursorRect.Y - innerScrollValue > keyboardYPosition)
+						move = cursorRect.Y - innerScrollValue - keyboardYPosition;
+				}
+
+				// Go up the hierarchy and look for other scrollViews until we reach the UIWindow
+				if (shouldContinue)
+				{
+					var tempScrollView = superScrollView.FindResponder<UIScrollView>();
+					UIScrollView? nextScrollView = null;
+
+					// set tempScrollView to next scrollable superview of superScrollView
+					while (tempScrollView is not null)
+					{
+						if (tempScrollView.ScrollEnabled)
+						{
+							nextScrollView = tempScrollView;
+							break;
+						}
+						tempScrollView = tempScrollView.FindResponder<UIScrollView>();
+					}
+
+					var shouldOffsetY = superScrollView.ContentOffset.Y - Math.Min(superScrollView.ContentOffset.Y, -move);
+
+					// the contentOffset.Y will change to shouldOffSetY so we can subtract the difference from the move
+					move -= (nfloat)(shouldOffsetY - superScrollView.ContentOffset.Y);
+
+					var newContentOffset = new CGPoint(superScrollView.ContentOffset.X, shouldOffsetY);
+
+					if (!superScrollView.ContentOffset.Equals(newContentOffset))
+					{
+						if (nextScrollView is null)
+						{
+							AnimateScroll(() =>
+							{
+								newContentOffset.Y += innerScrollValue;
+								innerScrollValue = 0;
+
+								if (View.FindResponder<UIStackView>() is not null)
+									superScrollView.SetContentOffset(newContentOffset, UIView.AnimationsEnabled);
+								else
+									superScrollView.ContentOffset = newContentOffset;
+							});
+						}
+
+						else
+						{
+							// add the amount we would have moved to the next scroll value
+							innerScrollValue += newContentOffset.Y;
+						}
+					}
+
+					lastView = superScrollView;
+					superScrollView = nextScrollView;
+				}
+
+				else
+				{
+					// if we did not get to scroll all the way, add the value to move
+					move += innerScrollValue;
+					break;
+				}
+			}
+		}
+
+		if (move >= 0)
+		{
+			rootViewOrigin.Y = (nfloat)Math.Max(rootViewOrigin.Y - move, Math.Min(0, -kbSize.Height - TextViewTopDistance));
+
+			if (RootController.Frame.X != rootViewOrigin.X || RootController.Frame.Y != rootViewOrigin.Y)
+			{
+				AnimateScroll(() =>
+				{
+					var rect = RootController.Frame;
+					rect.X = rootViewOrigin.X;
+					rect.Y = rootViewOrigin.Y;
+
+					RootController.Frame = rect;
+				});
+			}
+		}
+
+		else
+		{
+			var disturbDistance = rootViewOrigin.Y - TopViewBeginOrigin.Y;
+
+			if (disturbDistance <= 0)
+			{
+				rootViewOrigin.Y -= (nfloat)Math.Max(move, disturbDistance);
+
+				if (RootController.Frame.X != rootViewOrigin.X || RootController.Frame.Y != rootViewOrigin.Y)
+				{
+					AnimateScroll(() =>
+					{
+						var rect = RootController.Frame;
```

Comment:
Same, memory leak.

Reply:
Mea culpa, should have said increased memory pressure. We are creating an extra object with a reference to the capture elements. The gc will have extra work.

Reply:
Fixed!